PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -136,14 +155,90 @@ UpdateLoadedImage (
 
     // Save other binary blobs
     Index = 3;
-    while ((Index < MAX_MULTIBOOT_MODULE_NUMBER) && (Index < NumFiles)) {
+    while ((Index < MAX_EXTRA_FILE_NUMBER) && (Index < NumFiles)) {
+      // Update ACPI tables if we encounter an ACPI blob
+      if (File[Index].Name == SIGNATURE_32('A', 'C', 'P', 'I')) {
+          DEBUG ((DEBUG_INFO, "Loading boot image ACPI tables...\n"));
+          PlatformService = (PLATFORM_SERVICE *) GetServiceBySignature (PLATFORM_SERVICE_SIGNATURE);
+          if ((PlatformService != NULL) && (PlatformService->AcpiTableUpdate != NULL)) {
+            Status = PlatformService->AcpiTableUpdate (File[Index].Addr, File[Index].Size);
+            DEBUG ((DEBUG_INFO, "Updating ACPI table with boot image %d - %r\n", Index, Status));
+          }
+          FreeImageData (&File[Index]);
```

Comment:
Do we need to free? Does it create fagments?

Reply:
The call to FreeImageData() was present in the original implementation as well. It doesn't seem to create fragments but the ACPI table address seems to lie in the "System Memory" region. Should I remove it?

Reply:
ACPI tables should be marked as ACPI reclaim

Reply:
AcpiTableUpdate copies to ACPI Reclaim.